lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -827,15 +827,38 @@ def xid_or_ds_trans_id(three_d_secure_opts)
       end
 
       def add_stored_credential_data(parameters, credit_card_or_vault_id, options)
+        # Braintree has informed us that the stored_credential mapping may be incorrect
+        # In order to prevent possible breaking changes we will only apply the new logic if
+        # specifically requested. This will be the default behavior in a future release.
         return unless (stored_credential = options[:stored_credential])
 
-        parameters[:external_vault] = {}
-        if stored_credential[:initial_transaction]
-          parameters[:external_vault][:status] = 'will_vault'
+        add_external_vault(parameters, stored_credential)
+
+        if options[:new_stored_credential]
+          updated_stored_credentials(parameters, stored_credential)
         else
-          parameters[:external_vault][:status] = 'vaulted'
-          parameters[:external_vault][:previous_network_transaction_id] = stored_credential[:network_transaction_id]
+          classic_stored_credentials(parameters, stored_credential)
         end
+      end
+
+      def updated_stored_credentials(parameters, stored_credential)
+        case stored_credential[:reason_type]
+        when 'moto'
```

Comment:
How can `reason_type` be moto?

Reply:
Good catch, it can't! It was there before so I wonder if non-spreedly people pass it in? Do you know if we have a ticket to add `moto` to BT?

Reply:
Yes but it's set on `:sca_exemption`

Reply:
So `sca_exemption` is already part of Braintree. Do we also want to add if for `transaction_source` if customer sends `three_ds_exemption_type` as `moto`. Maybe that's for another ticket?

Reply:
Where do you see `sca_exemption` for Braintree? I don't see we support it now in AM

Reply:
https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/braintree_blue.rb#L690

Reply:
🤦🏽 it helps if i look at a somewhat new branch. I'm not sure if we should add it here for MOTO, I'll look through their docs more

Reply:
I do see their docs shows it but I think we can remove MOTO from now. I think I would make more sense to assign `transaction_source: 'moto'` if 3ds exemption type MOTO is passed. But we should probably confirm first and this could possible be a separate ticket.

Reply:
Cool I created a ticket and will remove it from here!